### PR TITLE
Refactoring 2014 06 24

### PR DESCRIPTION
**Pull request by thraxil**
some refactoring and test improvements.

### CODE REVIEW

File: wardenclyffe/main/views.py [hunk 1]
```diff
@@ -550,21 +529,10 @@ def s3_batch_upload(request, collection_id):
...
-            v.make_pull_from_s3_and_extract_metadata_operation(
-                key=key, user=request.user),
-            v.make_create_elastic_transcoder_job_operation(
-                key=key, user=request.user)]
-        if v.collection.audio:
-            o = v.make_audio_encode_operation(source_file.id, request.user)
-            v_operations.append(o)
+        v.make_source_file(key)
+        v.make_uploaded_source_file(key)
+        v_operations = v.initial_operations(key, request.user,
```
Comment (sdreher) [inline wardenclyffe/main/views.py]: The initial_operations function aligns with the existing code at [lines 496-505](https://github.com/ccnmtl/wardenclyffe/blob/master/wardenclyffe/main/views.py#L497). The refactored call at [line 483](https://github.com/ccnmtl/wardenclyffe/blob/0d5d4afd966b0179e22d18a65e2621539c055588/wardenclyffe/main/views.py#L483) seems correct.

However, the call to initial_operations here doesn't seem to match up with the existing code at [lines 560-69](https://github.com/ccnmtl/wardenclyffe/blob/master/wardenclyffe/main/views.py#L560) where a make_pull_from_s3_and_extract_metadata_operation and a make_create_elastic_transcoder_job_operation are completed in addition to a make_audio_encode_operation if audio.

File: wardenclyffe/mediathread/views.py [hunk 2]
```diff
@@ -16,32 +18,43 @@
 from django_statsd.clients import statsd
 
 
+class MediathreadAuthenticator(object):
```
Comment (sdreher) [inline wardenclyffe/mediathread/views.py]: nice encapsulation of the authentication logic.

File: wardenclyffe/mediathread/views.py [hunk 3]
```diff
@@ -124,70 +125,81 @@ def s3_upload(request):
...
+            url = mediathread_url(username)
+            credentials = None
+            if hasattr(settings, "MEDIATHREAD_CREDENTIALS"):
+                credentials = settings.MEDIATHREAD_CREDENTIALS
+            response = GET(url, credentials=credentials)
+            courses = loads(response)['courses']
+            courses = [dict(id=k, title=v['title'])
+                       for (k, v) in courses.items()]
+            courses.sort(key=lambda x: x['title'].lower())
+        except:
```
Comment (sdreher) [inline wardenclyffe/mediathread/views.py]: any way to resolve the bare except?

File: wardenclyffe/main/views.py (hunk 1 above)
Comment (thraxil) [inline wardenclyffe/main/views.py]: Yeah, I was naughty and didn't stick to pure refactoring there. As I put that one in, I noticed that the batch upload behavior still had the old (pre-full-S3) approach to audio encoding (I suspect no one has tried to do a batch upload of audio files recently, so no one encountered it), so I just switched it to the new approach while I was at it.

**Pull request merged**